code/__HELPERS/maths.dm

```diff
@@ -130,6 +130,27 @@ GLOBAL_LIST_INIT(sqrtTable, list(1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 4,
 	var/t = round((val - min) / d)
 	return val - (t * d)
 
+/proc/get_angle_of_incidence(face_angle, angle_in, auto_normalize = TRUE)
+	var/angle_in_s = angle_in
+	while(angle_in_s < 0)			//Don't round for optimal accuracy.

```

Comment:
in fact you could make a macro

    #define NORM_ROT(rot) (rot = rot%360; if rot < 0; rot +360)